deepsampler-junit/src/main/java/de/ppi/deepsampler/junit/LoadSamples.java

```diff
@@ -11,41 +11,54 @@
+ * the test class and test method. In most cases this will suffice, however you can change the path in detail:
+ * <ul>
+ *     <li>rootPath: The root path for relative packagePaths is by default './'. It can be changed, using the annotation
+ *     {@link SampleRootPath}.
+ *     <p>
+ *     The rootPath is ignored, if the file is loaded from the classpath.</li>
+ *     <li>packagePath: The path under the rootPath: If omitted, the package of the test class is used.</li>
+ *     <li>fileName: the concrete name of the JSON-file: If omitted, the name of the test class and the test method is used.</li>
+ * </ul>
  * <p>
- * This annotation must be used in combination with {@link UseSamplerFixture}.
+ * This annotation must be used in combination with @{@link UseSamplerFixture}.
  * <p>
- * The serialisation is done in two steps: First, the object may be converted to an abstract {@link de.ppi.deepsampler.persistence.model.PersistentBean}. This is done
- * to omit type information in the JSON-Files. Primitive types, arrays, {@link java.util.List}s and {@link java.util.Map}s are usually not converted to a
- * {@link de.ppi.deepsampler.persistence.model.PersistentBean}. They are passed to the second step unchanged. The second step is the JSON-serialisation by Jackson,
- * which may include some {@link com.fasterxml.jackson.databind.JsonSerializer}s
- * <p>
- * It is possible to register some extensions to customize the serialisation using the annotations @{@link UseJsonSerializer}
- * and {@link UseBeanConverterExtension}.
+ * It is possible to register some extensions to customize the deserialization using the annotations @{@link UseJsonDeserializer}
+ * and @{@link UseBeanConverterExtension}.
  */
 @Retention(RetentionPolicy.RUNTIME)
 @Target(ElementType.METHOD)
 public @interface LoadSamples {
 
+    /**
+     * Defines the parent folder of {@link LoadSamples#fileName()}. Default is the package of the test class.
+     * <p>
+     * If {@link LoadSamples#source()} is set to {@link FileSource#FILE_SYSTEM}, packagePath is treated as a simple
+     * path on the filesystem. If packagePath is a relative path, the root lies on './' by default. This root can be
+     * changed using @{@link SampleRootPath}.
+     *
+     * @return the parent path of {@link LoadSamples#fileName()}. Default is the package of the test class.
+     */
+    String packagePath() default AnnotationConstants.DEFAULT_VALUE_MUST_BE_CALCULATED;
 
     /**
-     * If this property is provided, the Sampler-file will be loaded from the local file system using the provided file name.
+     * The name of the json file without a path. Default is the name of the test class followed by the name of the test method.
      *
-     * @return the path to a Sampler-JSON-file.
+     * @return the name of the sample JSON file.
      */
-    String file() default "";
+    String fileName() default AnnotationConstants.DEFAULT_VALUE_MUST_BE_CALCULATED;
```

Comment:
packagename and filename are difficult to understand. They provide four option to configure. If someone wishes to use own filenames, they must provide a full qualified filename.

Reply:
We now have only one property which defines the complete file name including the path.